engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/NuKafkaRuntimeApp.scala

```diff
@@ -45,25 +51,35 @@ object NuKafkaRuntimeApp extends App with LazyLogging {
   }
   System.exit(exitCode)
 
+  private val akkaHttpCloseTimeout = 10 seconds
+
   private def runAfterActorSystemCreation(): Unit = {
     val scenarioInterpreter = RunnableScenarioInterpreterFactory.prepareScenarioInterpreter(scenario, runtimeConfig, liteKafkaJobData, system)
     Runtime.getRuntime.addShutdownHook(new Thread() {
       override def run(): Unit = {
+        logger.info("Closing http server")
+        if (server != null) Await.result(server.terminate(akkaHttpCloseTimeout), akkaHttpCloseTimeout)
```

Comment:
Why you moved it from runAfterActorSystemCreation to here (and extracted volataile variable because of that)? Is it some reason that it should be executed in this order?

Reply:
I think that you can remove `volatile private var server` field now. Can't you?

Reply:
... I've added one more slight detail - wrapped await with try-finally to properly handle edge case like bound exception